Std/Data/String/Lemmas.lean

```diff
@@ -499,6 +499,11 @@ theorem join_eq (ss : List String) : join ss = ⟨(ss.map data).join⟩ := go ss
 @[simp] theorem data_join (ss : List String) : (join ss).data = (ss.map data).join := by
   rw [join_eq]
 
+@[simp] theorem comp_length_data : List.length ∘ String.data = String.length := rfl
```

Comment:
I think this is more useful applied to an argument. Otherwise it is very unlikely to fire in practice.

Reply:
That's fair, I originally had it as a composition just so it can be used in `length_join` underneath, but I guess using `funext` is a pretty easy way to do that. Thanks!